src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -121,6 +123,12 @@ private MsQuicApi(NativeApi* vtable)
 
         static MsQuicApi()
         {
+            if (OperatingSystem.IsWindows() && !IsWindowsVersionSupported())
+            {
+                IsQuicSupported = false;
```

Comment:
Should we add trace message in case we need to debug it the field? Or is this going to be documented when we have QUIC API?

Reply:
Yes, I guess it makes sense to log current and min supported version here

Reply:
Hmm I guess there is no reliable way to get current Windows version, that's why #32575 was created in the first place. I will only log min supported version then

Reply:
`Environment.OSVersion` was changed to report current OS version as part of .NET 5. Should be safe to use if the desire is still there.